R/zchunk_L141.hfc_R_S_T_Y.R

```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Silence Package check notes
+    "." <- "year" <- "value" <- "GCAM_region_ID" <- "sector" <- "fuel" <- "service" <-
+    "IPCC_description" <- "agg_sector" <- "ISO_A3" <- "iso" <- "EDGAR_agg_sector" <-
+    "Non.CO2" <- "emissions" <- "Sector" <- "total" <- "share" <- "Species" <- "Emissions" <-
+    "SSP2_tot" <- "EDGAR_tot" <- "scaler" <- "supplysector" <- "subsector" <- "stub.technology" <-
+    "adj_emissions" <- "energy" <- "em_fact"
+
+    # Beginning processing and Mapping EDGAR HFC emissions to GCAM technologies
+
+    # Add column with Non.CO2 gas name
+    EDGAR_HFC125$Non.CO2 <- "HFC125"
+    EDGAR_HFC134a$Non.CO2 <- "HFC134a"
+    EDGAR_HFC143a$Non.CO2 <- "HFC143a"
+    EDGAR_HFC152a$Non.CO2 <- "HFC152a"
+    EDGAR_HFC227ea$Non.CO2 <- "HFC227ea"
+    EDGAR_HFC23$Non.CO2 <- "HFC23"
+    EDGAR_HFC236fa$Non.CO2 <- "HFC236fa"
+    EDGAR_HFC245fa$Non.CO2 <- "HFC245fa"
+    EDGAR_HFC32$Non.CO2 <- "HFC32"
+    EDGAR_HFC365mfc$Non.CO2 <- "HFC365mfc"
+    EDGAR_HFC43$Non.CO2 <- "HFC43"
+
+    # Process all F gasses files then combine and aggregate
+    # define function to formatt and process F gases, map to GCAM regions
+    format_F_Gas <- function(x){
+      x <- x %>%
+        left_join_error_no_match(EDGAR_sector %>% select(-IPCC_description), by = "IPCC") %>% # Add Edgar agg_sector
+        rename(EDGAR_agg_sector = agg_sector) %>% #rename agg_sector to EDGAR_agg_sector
+        mutate(iso = tolower(ISO_A3), ISO_A3 = NULL) %>% # convert to Edgar ISO
+        change_iso_code('rou', 'rom') %>% # Convert Romania iso code to pre-2002 value
+        left_join_error_no_match(iso_GCAM_regID,by = "iso") %>% # Map iso to GCAM region
+        select(GCAM_region_ID, iso, EDGAR_agg_sector,Non.CO2,matches(YEAR_PATTERN)) %>%
+        gather(year, emissions, -GCAM_region_ID,-iso,-EDGAR_agg_sector,-Non.CO2) %>%
+        mutate(year = as.numeric(year)) %>%
+        mutate(emissions = as.numeric(emissions))
+      return(x)
+    }
+    # combine all F gasses files into list
+    F_gases <- list(EDGAR_HFC125, EDGAR_HFC134a, EDGAR_HFC143a,
+                    EDGAR_HFC152a, EDGAR_HFC227ea, EDGAR_HFC23,
+                    EDGAR_HFC236fa, EDGAR_HFC245fa, EDGAR_HFC32,
+                    EDGAR_HFC365mfc, EDGAR_HFC43)
+
+    # Use function to convert datat to numeric values and combine to one tibble
+    # Map to GCAM countries and technologies
+    F_gases_formatted <- lapply(F_gases, format_F_Gas)
+
+    L141.EDGAR_HFC <- do.call(bind_rows,F_gases_formatted) %>%
```

Comment:
FYI this is unnecessary: `bind_rows` accepts a list of data frames along with explicitly-provided objects. I'll make this change.